src/assets/wise5/components/dialogGuidance/DialogGuidanceFeedbackRuleEvaluator.ts

```diff
@@ -114,11 +114,24 @@ export class DialogGuidanceFeedbackRuleEvaluator {
     return !this.evaluateTerm(termStack.pop(), response);
   }
 
-  private evaluateOrExpression(term1: string, term2: string, response: CRaterResponse): boolean {
-    return this.evaluateTerm(term1, response) || this.evaluateTerm(term2, response);
+  private evaluateTerm(term: string, response: CRaterResponse): boolean {
+    if (this.isHasKIScoreTerm(term)) {
+      return this.evaluateHasKIScoreTerm(term, response);
+    } else {
+      return this.evaluateIdeaTerm(term, response);
+    }
   }
 
-  private evaluateTerm(term: string, response: CRaterResponse): boolean {
+  private isHasKIScoreTerm(term: string): boolean {
+    return /hasKIScore\([\d]\)/.test(term);
```

Comment:
Should the regex use \d+ in case we have scores that are multiple digits in the future?

Reply:
Hmm I disagree here. I don't think we should code to handle a case in the future that may or may not happen.

On further thought, I think we should actually revert it to checking for [1-5] because those are the KI scores that we work with now, and we don't foresee any score values outside of those ranges in the near future. The code is more precise this way, so the reader can see that we only allow values 1-5 when checking for KI score. If the score range should change in the future, we can update the code then.